net/dns/publicdns/publicdns.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package publicdns contains mapping and helpers for working with
+// public DNS providers
+package publicdns
+
+import (
+	"sync"
+
+	"inet.af/netaddr"
+)
+
+var knownDoH = map[netaddr.IP]string{} // 8.8.8.8 => "https://..."
+var dohIPsOfBase = map[string][]netaddr.IP{}
+var populateOnce sync.Once
+
+// KnownDoH returns a map of well-known public DNS IPs to their DoH URL.
+// The returned map should not be modified.
+func KnownDoH() map[netaddr.IP]string {
+	populateOnce.Do(populate)
+	return knownDoH
+}
+
+// DoHIPsOfBase returns the public DNS provider associated with a given IP addr.
```

Comment:
text isn't quite right. kinda backwards.

... returns a map of DNS server IP addresses keyed by their DoH URL. It is the inverse of KnownDoH.